src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/DeclarationHandler.java

```diff
@@ -109,7 +109,8 @@ private Declaration handleNamespace(CPPASTNamespaceDefinition ctx) {
     lang.getScopeManager().enterScope(declaration);
     for (IASTNode child : ctx.getChildren()) {
       if (child instanceof IASTDeclaration) {
-        declaration.add(this.lang.getDeclarationHandler().handle((IASTDeclaration) child));
+        lang.getScopeManager()
```

Comment:
I think this is the only occurrence where adding a declaration is done by the scope manager and not by the translation manager. We should keep things consistent

Reply:
Yes, and consistently add them over the scope Manager. When a Declaration is processed, context knowledge is needed to know to which node a declaration has to be added. This is the exact purpose of a Manager whose job it is to keep track of scopes.
This is the only occurrence in `DeclarationHandler`, however, there are many more in the `DeclaratorHandler`.

Reply:
If we consistently add them via the scope manager that's fine for me :+1:

Reply:
Then at least `addDeclaration` should contain a javadoc containing information that this not only adds the declaration to the scope but also to the ast node containing that scope. I am still not a big fan of this because this way you hide the behaviour deep down in the scope manager but at least it should be documented and consistent.

Reply:
Update: Ok it looks like we are already doing it this way in the other handlers, looks like this one snuck by me :) Then let's do it this way - although I am not a fan, but please add documentation to the call.